src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -1482,6 +1482,10 @@ static QVariant fcnReplace( const QVariantList &values, const QgsExpressionConte
 
 static QVariant fcnRegexpReplace( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
 {
+  if ( ! values.at( 2 ).isValid() )
```

Comment:
```suggestion
  if ( ! values.at( 2 ).isNull() )
```

Better safe than sorry, an invalid is also null